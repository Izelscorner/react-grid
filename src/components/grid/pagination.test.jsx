import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Pagination from './pagination';

describe('<Pagination />', () => {
    configure({adapter: new Adapter()});
    let paginationHandler;
    let wrapper;
  
    beforeAll(() => {
      paginationHandler = jest.fn();
      wrapper = shallow(<Pagination 
        count={87} 
        currentPage={1} 
        size={10} 
        paginationHandler={paginationHandler} 
        paginationParam="page"  />);
    });
  
    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  
    it('page count should be 9', () => {
      const pages = wrapper.find('ul li');
      expect(pages).toHaveLength(9);
    });
  
    it('calls handleSearch on type', function () {
      const pageLast = wrapper.find('button').last();
      pageLast.simulate('click', {target: {value: pageLast.props().value}});
      expect(paginationHandler).toHaveBeenCalledWith({target: {value: 9}});
    });
});
