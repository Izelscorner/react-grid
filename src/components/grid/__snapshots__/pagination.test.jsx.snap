// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> matches snapshot 1`] = `
<nav
  aria-label="Page navigation example"
>
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item active"
      key="0"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={1}
      >
        1
      </button>
    </li>
    <li
      className="page-item "
      key="1"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={2}
      >
        2
      </button>
    </li>
    <li
      className="page-item "
      key="2"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={3}
      >
        3
      </button>
    </li>
    <li
      className="page-item "
      key="3"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={4}
      >
        4
      </button>
    </li>
    <li
      className="page-item "
      key="4"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={5}
      >
        5
      </button>
    </li>
    <li
      className="page-item "
      key="5"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={6}
      >
        6
      </button>
    </li>
    <li
      className="page-item "
      key="6"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={7}
      >
        7
      </button>
    </li>
    <li
      className="page-item "
      key="7"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={8}
      >
        8
      </button>
    </li>
    <li
      className="page-item "
      key="8"
    >
      <button
        className="page-link"
        name="page"
        onClick={[MockFunction]}
        value={9}
      >
        9
      </button>
    </li>
  </ul>
</nav>
`;
